cmake/make_dist/MakeDistCheck.cmake

```
@@ -36,43 +28,68 @@ function (RUN_DIST_CHECK PACKAGE_PREFIX EXT)
 
    file (MAKE_DIRECTORY ${BUILD_DIR} ${INSTALL_DIR})
 
+   # Ensure distcheck inherits polyfill library selection.
+   set(polyfill_flags "")
+   if (NOT "${CMAKE_CXX_STANDARD}" STREQUAL "")
+      list(APPEND polyfill_flags "-DCMAKE_CXX_STANDARD=${CMAKE_CXX_STANDARD}")
+   endif ()
+   if (NOT "${BOOST_ROOT}" STREQUAL "")
+      list(APPEND polyfill_flags "-DBOOST_ROOT=${BOOST_ROOT}")
+   endif ()
+   if (NOT "${BSONCXX_POLY_USE_MNMLSTC}" STREQUAL "")
+      list(APPEND polyfill_flags "-DBSONCXX_POLY_USE_MNMLSTC=${BSONCXX_POLY_USE_MNMLSTC}")
+   endif()
+   if (NOT "${BSONCXX_POLY_USE_STD_EXPERIMENTAL}" STREQUAL "")
+      list(APPEND polyfill_flags "-DBSONCXX_POLY_USE_STD_EXPERIMENTAL=${BSONCXX_POLY_USE_STD_EXPERIMENTAL}")
+   endif()
+   if (NOT "${BSONCXX_POLY_USE_SYSTEM_MNMLSTC}" STREQUAL "")
+      list(APPEND polyfill_flags "-DBSONCXX_POLY_USE_SYSTEM_MNMLSTC=${BSONCXX_POLY_USE_SYSTEM_MNMLSTC}")
+   endif()
+   if (NOT "${BSONCXX_POLY_USE_BOOST}" STREQUAL "")
+      list(APPEND polyfill_flags "-DBSONCXX_POLY_USE_BOOST=${BSONCXX_POLY_USE_BOOST}")
+   endif()
+   if (NOT "${BSONCXX_POLY_USE_STD}" STREQUAL "")
+      list(APPEND polyfill_flags "-DBSONCXX_POLY_USE_STD=${BSONCXX_POLY_USE_STD}")
+   endif()
+
+   execute_process_and_check_result (
+      COMMAND ${CMAKE_COMMAND} -E echo "Configuring distcheck with CMake flags: ${polyfill_flags}"
+      WORKING_DIRECTORY .
+      ERROR_MSG "Failed to echo polyfill flags"
+   )
+
    execute_process_and_check_result (
       COMMAND ${CMAKE_COMMAND}
+         -S ../${PACKAGE_PREFIX}
+         -B .
          -DCMAKE_BUILD_TYPE=Release
          -DMONGOCXX_ENABLE_SLOW_TESTS=ON
          -DCMAKE_PREFIX_PATH=${CMAKE_PREFIX_PATH}
          -DCMAKE_INSTALL_PREFIX=../${INSTALL_DIR}
-         -DMONGOCXX_ENABLE_SLOW_TESTS=ON
-         ../${PACKAGE_PREFIX}
+         ${polyfill_flags}
+
       WORKING_DIRECTORY ${BUILD_DIR}
       ERROR_MSG "CMake configure command failed."
    )
 
    # Run make in the build directory
-   if (DEFINED ENV{DISTCHECK_BUILD_OPTS})
-      set (build_opts $ENV{DISTCHECK_BUILD_OPTS})
-   else ()
-      set (build_opts "-j 8")
-   endif ()
    separate_arguments (build_opts)
    execute_process_and_check_result (
-      COMMAND ${MY_CMAKE_COMMAND} ${MAKE_COMMAND} ${build_opts}
+      COMMAND ${CMAKE_COMMAND} --build .
```

```suggestion
      COMMAND ${CMAKE_COMMAND} --build . --parallel
```

If `--parallel` is given with no argument value, CMake will infer the parallelism level automatically (may be useful in other contexts, but is a useful shorthand in general).

---

Confirming, is inferring parallelism via `--parallel` preferable/different to using the value of the `CMAKE_BUILD_PARALLEL_LEVEL` environment variable (which AFAIK should be visible/inherited by the `execute_process()` commands given the changes in this PR)?

---

Confirmed: `CMAKE_BUILD_PARALLEL_LEVEL` suffices.